shared/libraries/shooter.h

```diff
@@ -25,34 +25,73 @@
 
 namespace control {
 
-typedef enum{
-	MOTOR_Left = 0,
-	MOTOR_Rright = 1,
-	MOTOR_Bullet = 2,
-} shooter_motor_t;
-
-typedef enum{
-	HERO = 0,
-	STANDARD = 1,
-	ENGINEER = 2,
-	AERIAL = 3,
-	SENTRY = 4,
-	DART = 5,
-	RADAR = 6
-} robot_type_t;
+/**
+ * @brief structure used when shooter instance is initialized
+ */
+typedef struct {
+  bool acc_using_can_motor;					 /* if accelerate motors are using CAN protocal  */
+  MotorCANBase* left_acc_can_motor;	 /* CAN motor instance of left accelerate motor  */
+  MotorCANBase* right_acc_can_motor; /* CAN motor instance of right accelerate motor */
+  MotorPWMBase* left_acc_pwm_motor;	 /* PWM motor instance of left accelerate motor  */
+  MotorPWMBase* right_acc_pwm_motor; /* PWM motor instance of right accelerate motor */
+  bool left_acc_motor_invert;				 /* if left accelrate motor is inverted					 */
+  bool right_acc_motor_invert;			 /* if right accelrate motor is inverted				 */
+  ServoMotor* load_servo;            /* servomotor instance of load motor            */
+  float acc_Kp;                      /* Kp of pid controlling accelerate motor speed */ 
+  float acc_Ki;                      /* Ki of pid controlling accelerate motor speed */
+  float acc_Kd;                      /* Kd of pid controlling accelerate motor speed */
+  float load_step_angle;             /* step size of loading motor, in [rad]         */
+} shooter_t;
 
+/**
+ * @brief wrapper class for shooter
+ */
 class Shooter {
 public:
-    Shooter(robot_type_t robot_type, const int* motor_id, const float* fire_pid, const float* load_pid);
-    void Fire(float speed);
-    void Load(float speed);
+  /**
+   * @brief constructor for Shooter instance
+   * 
+   * @param shooter structure that used to initialize gimbal, refer to type shooter_t
+   */
+  Shooter(shooter_t shooter);
+
+  /**
+   * @brief set the speed of accelerating motors
+   * 
+   * @param speed 
+   */
+  void SetAccSpeed(float speed);
```

Comment:
I call the motors that used to accelerate the projectiles "accelerating motors", and I abbr. it in the function as SetAccSpeed. However, SetSpeed will still be confusing since there are two sets of motors controlled by the Shooter class. One group is accelerating motors as I just mentioned, and the other group is the load motor. SetSpeed would be ambiguous because it is not clear that the speed of which set of motors are changed.